bigquery/google/cloud/bigquery/job.py

```diff
@@ -1999,6 +2001,13 @@ def default_dataset(self):
         to use for unqualified table names in the query or :data:`None` if not
         set.
 
+        The ``dataset`` setter accepts:
+
+        - a :class:`~google.cloud.bigquery.dataset.DatasetReference`, or
+        - a :class:`str` of the fully-qualified dataset ID in standard SQL
+          format. The value must included a project ID and dataset ID
+          separated by ``.``.
```

Comment:
Would be nice to have an example here (e.g, `your-project.your_dataset`)

Reply:
Done in https://github.com/googleapis/google-cloud-python/pull/7560/commits/3ab95d4ad7da8b2be27171716e294f024882536d.